src/executionTimingsReporter.ts

```diff
@@ -0,0 +1,128 @@
+import { HighResTimer, calculateElapsedTime } from "./utils";
+import { EventEmitter } from "events";
+
+export type TimingsReport = {[stepName: string]: [number] | number};
+
+const stepStartedEventName = 'stepStarted';
+const milestoneReachedEventName = 'milestoneReached';
+
+interface StepStartedEventArguments {
+    stepName: string;
+}
+
+interface MilestoneReachedEventArguments {
+    milestoneName: string;
+}
+
+export interface ObservableEvents {
+    Events: EventEmitter;
+}
+
+export class StepStartedEventsEmitter extends EventEmitter {
```

Comment:
I feel that this class already has a verb `start` in it. So repeat that in the class name might be a little off?  Since it's not all about `start` (correct me if I am wrong).

Would it be more like a `StepProgressEventsEmitter` ?

Reply:
Done